compiler/noirc_evaluator/src/acir/tests/mod.rs

```diff
@@ -926,3 +927,208 @@ fn do_not_overflow_with_constant_constrain_neq() {
+        _ => format!("result = {} lhs, rhs", op),
+    };
+
+    if output {
+        (
+            format!(
+                "
+            {src}
+        return result
+        }}"
+            ),
+            true,
+        )
+    } else {
+        (
+            format!(
+                "
+            {src}
+        return
+        }}"
+            ),
+            false,
+        )
+    }
+}
+
+// Execute a simple operation for each operators
+// The operation is executed from SSA IR using ACVM after acir-gen
+// and also directly on the SSA IR via constant folding, by hardcoding the values
+// via get_constant_main_src() prefix.
+fn test_operators(
+    // The list of operators to test
+    operators: &[&str],
+    // the type of the input values
+    typ: &str,
+    // the values of the inputs
+    inputs: &[FieldElement],
+) {
+    let main = get_main_src(typ);
+    let num_type = match typ.chars().next().unwrap() {
+        'F' => NumericType::NativeField,
+        'i' => NumericType::Signed { bit_size: typ[1..].parse().unwrap() },
+        'u' => NumericType::Unsigned { bit_size: typ[1..].parse().unwrap() },
+        _ => unreachable!("invalid numeric type"),
+    };
+    let inputs_int = Value::array_from_iter(inputs.iter().cloned(), num_type).unwrap();
+    let inputs =
+        inputs.iter().enumerate().map(|(i, f)| (Witness(i as u32), *f)).collect::<BTreeMap<_, _>>();
+    let len = inputs.len() as u32;
+    let initial_witness = WitnessMap::from(inputs);
+
+    for op in operators {
+        let (src, with_output) = generate_test_instruction_from_operator(op);
+        let output = if with_output { Some(Witness(len)) } else { None };
+        let ssa = Ssa::from_str(&(main.to_owned() + &src)).unwrap();
+        // ssa execution
+        let result = ssa.interpret(vec![inputs_int.clone()]);
+        // acir execution
+        let result2 = execute_ssa(ssa, initial_witness.clone(), output.as_ref());
+
+        match (result, result2) {
+            // Both execution failed, so it is the same behavior, as expected.
+            (Err(_), (ACVMStatus::Failure(_), _)) => (),
+            // Both execution succeeded and output the same value
+            (Ok(ssa_result), (ACVMStatus::Solved, acvm_result)) => {
+                let result = if let Some(result) = ssa_result.first() {
+                    result.as_numeric().map(|v| v.convert_to_field())
+                } else {
+                    None
+                };
+                assert_eq!(result, acvm_result);
+            }
+            _ => panic!("ssa and acvm execution should have the same result"),
+        }
+    }
+}
+
+#[test]
+fn test_binary_on_field() {
+    // Test the following Binary operation on Fields
+    let operators = [
+        "add",
+        "sub",
+        "mul",
+        "div",
+        "eq",
+        // Bitwise operations are not allowed on field elements
+        // SSA interpreter will emit an error but not ACVM
+        // "and",
+        // "xor",
```

Comment:
We can follow-up with expected error tests that would have corresponding issues.